tests/server/conftest.py

```diff
@@ -117,8 +117,10 @@ def inner(
         server: Union[str, OptimadeTestClient] = "regular",
     ):
         response = get_good_response(request, server)
-
-        response_ids = [struct["id"] for struct in response["data"]]
+        if type(response["data"]) == dict:
```

Comment:
```suggestion
        if isinstance(response["data"], dict):
```

Better to use `isinstance`: it is faster and can handle subclasses.